packages/eas-cli/src/credentials/ios/api/GraphqlClient.ts

```diff
@@ -0,0 +1,233 @@
+    return await AppQuery.byFullNameAsync(projectFullName);
+  }
+
+  public async createOrUpdateIosAppBuildCredentialsAsync(
+    appLookupParams: AppLookupParams,
+    {
+      appleTeam,
+      appleAppIdentifierId,
+      iosDistributionType,
+      appleProvisioningProfileId,
+      appleDistributionCertificateId,
+    }: {
+      appleTeam: AppleTeam;
+      appleAppIdentifierId: string;
+      iosDistributionType: IosDistributionType;
+      appleProvisioningProfileId: string;
+      appleDistributionCertificateId: string;
+    }
+  ): Promise<IosAppBuildCredentials> {
+    const projectFullName = formatProjectFullName(appLookupParams);
+    const maybeIosAppCredentials = await IosAppCredentialsQuery.withBuildCredentialsByAppIdentifierIdAsync(
+      projectFullName,
+      { appleAppIdentifierId, iosDistributionType }
+    );
+    let iosAppCredentials: IosAppCredentials;
+    if (maybeIosAppCredentials) {
+      iosAppCredentials = maybeIosAppCredentials;
+    } else {
+      const { account } = appLookupParams;
+      const app = await this.getAppAsync(appLookupParams);
+      await IosAppCredentialsMutation.createIosAppCredentialsAsync(
+        { appleTeamId: appleTeam.id },
+        app.id,
+        account.id
+      );
+      iosAppCredentials = nullthrows(
+        await IosAppCredentialsQuery.withBuildCredentialsByAppIdentifierIdAsync(projectFullName, {
+          appleAppIdentifierId,
+          iosDistributionType,
+        })
+      );
+    }
+
+    const iosAppBuildCredentials = iosAppCredentials.iosAppBuildCredentialsArray?.[0];
+    if (!iosAppBuildCredentials) {
+      return await IosAppBuildCredentialsMutation.createIosAppBuildCredentialsAsync(
+        {
+          iosDistributionType,
+          distributionCertificateId: appleDistributionCertificateId,
+          provisioningProfileId: appleProvisioningProfileId,
+        },
+        iosAppCredentials.id
+      );
+    } else {
+      await IosAppBuildCredentialsMutation.setDistributionCertificateAsync(
+        iosAppBuildCredentials.id,
+        appleDistributionCertificateId
+      );
+      return await IosAppBuildCredentialsMutation.setProvisioningProfileAsync(
+        iosAppBuildCredentials.id,
+        appleProvisioningProfileId
+      );
+    }
+  }
+
+  public async createOrGetExistingAppleTeamAsync(
+    { account }: AppLookupParams,
+    { appleTeamIdentifier, appleTeamName }: { appleTeamIdentifier: string; appleTeamName: string }
+  ): Promise<AppleTeam> {
+    const appleTeam = await AppleTeamQuery.byAppleTeamIdentifierAsync(
+      account.id,
+      appleTeamIdentifier
+    );
+    if (appleTeam) {
+      return appleTeam;
+    } else {
+      return await AppleTeamMutation.createAppleTeamAsync(
+        { appleTeamIdentifier, appleTeamName },
+        account.id
+      );
+    }
+  }
+
+  public async createOrGetExistingAppleAppIdentifierAsync(
+    { account, bundleIdentifier }: AppLookupParams,
```

Comment:
does this also account for wildcard bundle identifiers (ie) `com.*` ? Or is this only for explicit bundle identifiers (ie) `com.foo`?

Reply:
Yes, it does. 
- https://github.com/expo/universe/blob/master/server/www/src/data/entities/credentials/AppleAppIdentifierEntity.ts#L215 - this is used in the mutation
- https://github.com/expo/universe/blob/master/server/www/src/data/entities/credentials/AppleAppIdentifierEntity.ts#L90 - this is used in the query